torch/csrc/api/src/mps.cpp

```diff
@@ -25,9 +21,22 @@ void manual_seed(uint64_t seed) {
 }
 
 void synchronize() {
-  TORCH_CHECK(is_available(), "No MPS devices are available");
   at::detail::getMPSHooks().deviceSynchronize();
 }
 
+void commit() {
+  at::detail::getMPSHooks().commitStream();
+}
```

Comment:
Why not move this implementation to header and say that function is `inline`?

Reply:
I wanted the `MPSHooks` remain in the libtorch binary and not expose it.
But if you think that wouldn't make any problem with external users that only link their project against libtorch and use torch/mps.h header only (i.e., users who don't want to depend on any other torch headers like ATen/Context.h), then I can make those APIs static inline and expose MPSHooks in the mps.h header.

Reply:
@malfet , what do you think? are there any concerns.